src/v/cluster/topics_frontend.cc

```diff
@@ -498,4 +503,76 @@ ss::future<result<model::offset>> topics_frontend::stm_linearizable_barrier(
     });
 }
 
+ss::future<std::vector<topic_result>> topics_frontend::create_partitions(
+  std::vector<create_partititions_configuration> partitions,
+  model::timeout_clock::time_point timeout) {
+    auto r = co_await stm_linearizable_barrier(timeout);
+    if (!r) {
+        std::vector<topic_result> results;
+        results.reserve(partitions.size());
+        std::transform(
+          partitions.begin(),
+          partitions.end(),
+          std::back_inserter(results),
+          [err = r.error()](const create_partititions_configuration& cfg) {
+              return make_error_result(cfg.tp_ns, err);
+          });
```

Comment:
was the intention to return something here? it looks like `results` isn't used after it is filled, either.